src/overlays/gamestates/ovl_file_choose/z_file_choose_NES.c

```diff
@@ -4,96 +4,2362 @@
+                j = D_80814644[this->buttonIndex];
+            } else if ((this->configMode == 0x19) || (this->configMode == 0xC)) {
+                j = D_8081464C[this->buttonIndex];
+            } else {
+                j = D_80814638[this->buttonIndex];
+            }
+        } else {
+            j = D_80814650[this->confirmButtonIndex];
+        }
+
+        this->windowContentVtx[i + 0].v.ob[0] = this->windowContentVtx[i + 2].v.ob[0] = this->windowPosX - 0xA;
+
+        this->windowContentVtx[i + 1].v.ob[0] = this->windowContentVtx[i + 3].v.ob[0] =
+            this->windowContentVtx[i + 0].v.ob[0] + 0x48;
+
+        this->windowContentVtx[i + 0].v.ob[1] = this->windowContentVtx[i + 1].v.ob[1] =
+            this->windowContentVtx[j].v.ob[1] + 4;
+
+        this->windowContentVtx[i + 2].v.ob[1] = this->windowContentVtx[i + 3].v.ob[1] =
+            this->windowContentVtx[i + 0].v.ob[1] - 0x18;
+
+        this->windowContentVtx[i + 1].v.tc[0] = this->windowContentVtx[i + 3].v.tc[0] = 0x900;
+
+        this->windowContentVtx[i + 2].v.tc[1] = this->windowContentVtx[i + 3].v.tc[1] = 0x300;
+    }
+
+    this->windowContentVtx[i + 4].v.ob[0] = this->windowContentVtx[i + 6].v.ob[0] = this->windowPosX + 0x3A;
+
+    this->windowContentVtx[i + 5].v.ob[0] = this->windowContentVtx[i + 7].v.ob[0] =
+        this->windowContentVtx[i + 4].v.ob[0] + 0x80;
+
+    this->windowContentVtx[i + 4].v.ob[1] = this->windowContentVtx[i + 5].v.ob[1] =
+        this->windowContentVtx[D_80814638[this->warningButtonIndex]].v.ob[1];
+
+    this->windowContentVtx[i + 6].v.ob[1] = this->windowContentVtx[i + 7].v.ob[1] =
+        this->windowContentVtx[i + 4].v.ob[1] - 0x10;
+
+    this->windowContentVtx[i + 5].v.tc[0] = this->windowContentVtx[i + 7].v.tc[0] = 0x1000;
+}
+#else
+void FileSelect_SetWindowContentVtx(GameState* thisx);
+#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_file_choose/FileSelect_SetWindowContentVtx.s")
+#endif
+
+u16 D_80814654[] = {
+    0x88,
+    0x194,
+    0x2A0,
+};
+TexturePtr sFileSelRemainsTextures[] = {
+    gFileSelOdolwasRemainsTex,
+    gFileSelGohtsRemainsTex,
+    gFileSelGyorgsRemainsTex,
+    gFileSelTwinmoldsRemainsTex,
+};
+TexturePtr sFileSelDayENGTextures[] = {
+    gFileSelFirstDayENGTex,
+    gFileSelFirstDayENGTex,
+    gFileSelSecondDayENGTex,
+    gFileSelFinalDayENGTex,
+};
+TexturePtr sFileSelHeartPieceTextures[] = {
+    gFileSel0QuarterHeartENGTex,
+    gFileSel1QuarterHeartENGTex,
+    gFileSel2QuarterHeartENGTex,
+    gFileSel3QuarterHeartENGTex,
+};
+static TexturePtr sHeartTextures[2][5] = {
+    {
+        gHeartEmptyTex,
+        gHeartQuarterTex,
+        gHeartHalfTex,
+        gHeartThreeQuarterTex,
+        gHeartFullTex,
+    },
+    {
+        gDefenseHeartEmptyTex,
+        gDefenseHeartQuarterTex,
+        gDefenseHeartHalfTex,
+        gDefenseHeartThreeQuarterTex,
+        gDefenseHeartFullTex,
+    },
+};
+u8 sHealthToQuarterHeartCount[] = {
+    0, 1, 1, 1, 1, 1, 2, 2, 2, 2, 2, 3, 3, 3, 3, 3,
```

Comment:
Does this match up with a similar array in parameter/lifemeter somewhere?

Reply:
lifemeter just has the 16 textures directly instead of only having a subset of textures and using a 2D array with indices to know which to use:
https://github.com/zeldaret/mm/blob/cc6cc8da96ceaad7b1bfa318e5e3c422b2a89ba7/src/code/z_lifemeter.c#L19-L33

Reply:
Odd, why are the array names in PascalCase?

Reply:
Something to cleanup later